catalog/spatial/registry/registry-federation-admin-impl/src/main/java/org/codice/ddf/registry/federationadmin/impl/FederationAdmin.java

```diff
@@ -0,0 +1,699 @@
+            String rightNow = now.toString();
+
+            for (SlotType1 slot : nodeInfo.getSlot()) {
+                if (slot.getName()
+                        .equals(RegistryConstants.XML_LIVE_DATE_NAME)) {
+                    liveDateFound = true;
+                } else if (slot.getName()
+                        .equals(RegistryConstants.XML_LAST_UPDATED_NAME)) {
+                    ValueListType valueList = EbrimConstants.RIM_FACTORY.createValueListType();
+                    valueList.getValue()
+                            .add(rightNow);
+                    slot.setValueList(EbrimConstants.RIM_FACTORY.createValueList(valueList));
+                    lastUpdatedFound = true;
+                }
+            }
+
+            if (!liveDateFound) {
+                SlotType1 liveDate = slotHelper.create(RegistryConstants.XML_LIVE_DATE_NAME,
+                        rightNow,
+                        DatatypeConstants.DATETIME.toString());
+
+                nodeInfo.getSlot()
+                        .add(liveDate);
+            }
+
+            if (!lastUpdatedFound) {
+                SlotType1 lastUpdated = slotHelper.create(RegistryConstants.XML_LAST_UPDATED_NAME,
+                        rightNow,
+                        DatatypeConstants.DATETIME.toString());
+
+                nodeInfo.getSlot()
+                        .add(lastUpdated);
+            }
+        }
+    }
+
+    private void configureMBean() {
+        mbeanServer = ManagementFactory.getPlatformMBeanServer();
+
+        try {
+            objectName = new ObjectName(FederationAdminMBean.OBJECT_NAME);
+        } catch (MalformedObjectNameException e) {
+            LOGGER.warn("Exception while creating object name: " + FederationAdminMBean.OBJECT_NAME,
+                    e);
+        }
+
+        try {
+            try {
+                mbeanServer.registerMBean(new StandardMBean(this, FederationAdminMBean.class),
+                        objectName);
+            } catch (InstanceAlreadyExistsException e) {
+                mbeanServer.unregisterMBean(objectName);
+                mbeanServer.registerMBean(new StandardMBean(this, FederationAdminMBean.class),
+                        objectName);
+            }
+        } catch (Exception e) {
+            LOGGER.error("Could not register mbean.", e);
+        }
+    }
+
+    public void destroy() {
+        try {
+            if (objectName != null && mbeanServer != null) {
+                mbeanServer.unregisterMBean(objectName);
+            }
+        } catch (Exception e) {
+            LOGGER.warn("Exception un registering mbean: ", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void init() {
+
+        String registryFieldsPath = String.format("%s%s%s%s%s",
+                System.getProperty(KARAF_ETC),
+                File.separator,
+                REGISTRY_CONFIG_DIR,
+                File.separator,
+                REGISTRY_FIELDS_FILE);
+
```

Comment:
Shouldn't you do something like this instead of using File.separator:
` String karafEtc = System.getProperty(KARAF_ETC);`
`Paths.get(karafEtc, REGISTRY_CONFIG_DIR, REGISTRY_FIELDS_FILE);`?